markdownlint/TOP007_useMarkdownLinks/TOP007_useMarkdownLinks.js

```diff
@@ -0,0 +1,74 @@
+function extractHref(link) {
+  // https://regexr.com/7v3ci to test this regex
+  return link.match(/(?<=href=("|'))[^"']+/)?.[0];
+}
+
+function extractLinkText(link) {
+  const textStart = link.indexOf(">") + 1;
+  const textEnd = link.lastIndexOf("<");
+  return link.substring(textStart, textEnd);
+}
+
+module.exports = {
+  names: ["TOP007", "use-markdown-links"],
+  description:
+    "Links used to navigate to external content or other landmarks in the page should use markdown links instead of HTML anchor tags.",
+  tags: ["links", "html"],
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP007.md"
+  ),
+  function: function TOP007(params, onError) {
+    const fencesLineRanges = params.tokens
+      .filter((token) => token.type === "fence")
+      .map((token) => token.map);
+
+    const isWithinFencedCodeBlock = (lineNumber) => {
+      return fencesLineRanges.some((range) => range[0] < lineNumber && lineNumber < range[1]);
+    };
+
+    const anchorsToFlag = params.lines.reduce((anchors, currentLine, index) => {
+      if (
+        !currentLine.includes("<a ") ||
+        !currentLine.includes("</a>") ||
+        isWithinFencedCodeBlock(index)
+      ) {
+        return anchors;
+      }
+
+      // https://regexr.com/7v3bb to test this regex
+      const nonCodepenAnchorsWithHrefRegex = /(?<!`)<a\s[^>]*href=[^>]+>.+?<\/a>(?!`)/g;
+      const anchorsInCurrentLine = currentLine.match(nonCodepenAnchorsWithHrefRegex);
+
+      anchorsInCurrentLine?.forEach((anchor) => {
+        if (anchor.includes("codepen")) {
+          return;
+        }
```

Comment:
One issue I'd see with this is that if we were to link to Codepen for whatever reason, outside of an embed, that link would be skipped.

What's interesting is that our flexbox axes lesson has an embed that uses Markdown links instead of HTML tags (see the [Axes section](https://www.theodinproject.com/lessons/foundations-axes#axes), first embed there; [axes lesson on GitHub](https://github.com/TheOdinProject/curriculum/blob/main/foundations/html_css/flexbox/flexbox-axes.md#axes)), and the embed actually looks to work fine. So despite all this work (which is great and 100% appreciated), I wonder if the easiest solution would be to just remove the `"a"` element from our allowed list of elements for rule MD033.

The more involved solution would be to ignore `<a>` elements that are children of `p.codepen` elements and keeping this rule.